quinn-proto/src/connection/mod.rs

```diff
@@ -1791,6 +1792,17 @@ impl Connection {
                 lost_packets, size_of_lost_packets
             );
 
+            // Packets sent before this time are deemed lost.
+            // We avoid computing this value above, since it's possible for this to panic
+            // if the `loss_delay` value internally stores a bigger `Duration` than the
+            // `Duration` that's stored inside the `Instant`, because some platforms may
+            // implement the `Instant` with a counter relative to system or even process
+            // startup (Wasm is one such case).
+            // If we're at this point, then it must be possible to have instants that are
+            // longer ago than `loss_delay` (see the `packet_too_old` computation
+            // above).
```

Comment:
This invariant isn't at all obvious to me. Can you explain more how/why this is the case?

Couldn't we do `now.checked_duration_since(info.time_sent)` and key off the returning `Option` instead, to avoid the `unwrap()` here?

Reply:
> Couldn't we do `now.checked_duration_since(info.time_sent)` and key off the returning `Option` instead, to avoid the `unwrap()` here?

I thought so at first too, but then realized that `now - info.time_sent` != `now - loss_delay`.
So this would change the qlog events emitted.

---

> This invariant isn't at all obvious to me. Can you explain more how/why this is the case?

I checked this again, which made me notice this invariant isn't quite correct.

Should we rather skip qlog events (and ideally log a warning?) if `now.checked_sub(loss_delay)` ends up being `None`?

Reply:
> I checked this again, which made me notice this invariant isn't quite correct.
> 
> Should we rather skip qlog events (and ideally log a warning?) if `now.checked_sub(loss_delay)` ends up being `None`?

Yes, that sounds better.